extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -36,7 +38,49 @@ export class AmpExperiment extends AMP.BaseElement {
       toggle(this.element, false);
       return;
     }
-    dev.info(EXPERIMENT, 'amp-experiment is on');
+
+    const config = this.getConfig_();
+    const results = {};
+    this.experimentVariants = Promise.all(
+        Object.keys(config).map(experimentName => {
+          return this.getVariantAllocation_(config[experimentName])
+              .then(variantName => {
+                if (variantName) {
+                  results[experimentName] = variantName;
+                }
+              });
+        })).then(() => results);
+    this.experimentVariants.then(this.addToBody_());
+  }
+
+  getConfig_() {
+    const children = this.element.children;
+    user.assert(
```

Comment:
I'm not sure if it's worth of the payoff of having more characters in the final JS. I think the current error message is already quite clear.